src/util/internalCallTree.js

```diff
@@ -189,6 +194,24 @@ function includeVariableDeclaration (tree, step, sourceLocation, scopeId) {
       }
     })
   }
+  var functionDefinition = resolveFunctionDefinition(tree, step, previousSourceLocation)
+  if (functionDefinition && newLocation && traceHelper.isJumpDestInstruction(tree.traceManager.trace[step - 1])) {
+    // means: the previous location was a function definition && JUMPDEST
```

Comment:
What happens if you return from a function call? I think you should also check whether the previous jump was a "jump in".

Reply:
we check if  `step - 1` is a JumpDest. In case of returning from a function call, `functionDefinition` would be null (means the current source location is not a function).

Reply:
What about the following code:
```
function f() {
  uint x = 2;
  x = 2 * g();
}
```
At the point where we return from the call to `g()`, we are again in code that has a function definition, it is a new location, we are just after a jump dest, but this is still not the beginning of the function.  The only different to the jump into the code of `f` is that in that case, the previous jump was a jump "into".

Reply:
your example is working ok ... but I think i know why this confusion: the `functionDefinition` variable is not the source location inside a `functionDefinition` . this variable contains the source location of an entire `functionDefinition` (if applicable) or null.

Reply:
Ok, I see. So this seems to work for most of the cases. I'm still not convinced that this works all the time, but we can merge it regardless :-)